x/evm/types/statedb.go

```diff
@@ -785,8 +785,10 @@ func (csdb *CommitStateDB) getStateObject(addr ethcmn.Address) (stateObject *sta
 		return nil
 	}
 
+	balance := csdb.bankKeeper.GetBalance(csdb.ctx, acc.GetAddress(), emint.DenomDefault)
+
 	// insert the state object into the live set
-	so := newStateObject(csdb, acc)
+	so := newStateObject(csdb, acc, balance.Amount)
```

Comment:
This is the source of the bug. Everytime the state was retrieved it was being initialized with 0 balance. @ColinSchwarz @austinabell @noot @austinabell